tests/core/test_standard_readable.py

```diff
@@ -0,0 +1,149 @@
+from inspect import ismethod
+from typing import List, get_type_hints
+from unittest.mock import MagicMock
+
+import pytest
+from bluesky.protocols import HasHints
+
+from ophyd_async.core import ConfigSignal, HintedSignal, StandardReadable
+from ophyd_async.core.device import Device
+from ophyd_async.core.signal import SignalR
+from ophyd_async.core.sim_signal_backend import SimSignalBackend
+from ophyd_async.protocols import AsyncConfigurable, AsyncReadable, AsyncStageable
+
+
+def test_standard_readable_hints():
+    sr = StandardReadable()
+
+    assert sr.hints == {}
+
+    hint1 = MagicMock()
+    hint1.hints = {"fields": ["abc"], "dimensions": [(["f1", "f2"], "s1")]}
+
+    hint2 = MagicMock()
+    hint2.hints = {"fields": ["def", "ghi"]}
+
+    hint3 = MagicMock()
+    hint3.hints = {"fields": ["jkl"], "gridding": "rectilinear_nonsequential"}
+
+    sr._has_hints = (hint1, hint2, hint3)
```

Comment:
Why not `sr.add_readables([hint1, hint2, hint3])`?

Reply:
At the time it was avoiding testing more code than necessary for this. However, given that `add_readables` has now got much simpler as the logic to create the hints has been moved, I've changed this as suggested.